js/ui/grid_core/ui.grid_core.column_headers.js

```diff
@@ -190,6 +256,22 @@ module.exports = {
                     }
 
                     that.callBase.apply(that, arguments);
+
+                    if(!that._isLegacyKeyboardNavigation()) {
+                        that._restoreLastFocusedElement();
+                    }
+                },
+
+                _restoreLastFocusedElement: function() {
+                    var $table = this.element(),
+                        $firstCell = $table.find(`tr.${HEADER_ROW_CLASS} > td`).first();
+
+                    eventsEngine.on($firstCell, "focus", e => {
```

Comment:
Why do we need to subscribe on focus event instead of direct lastActionElement focusing?

Reply:
Here we subscribe on focus event for the first header element because it will be focused first when focus got in grid. And after that we restore focus for the last focused element